packages/@webex/plugin-cc/test/unit/spec/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { WebSocketManager } from '../../../../../../src/services/core/WebSocket/WebSocketManager';
+import { WebexSDK, SubscribeRequest } from '../../../../../../src/types';
+import { SUBSCRIBE_API, WCC_API_GATEWAY } from '../../../../../../src/services/constants';
+
+jest.mock('../../../../../../src/services/core/HttpRequest');
+jest.mock('../../../../../../src/logger-proxy', () => ({
+  __esModule: true,
+  default: {
+    logger: {
+      log: jest.fn(),
+      error: jest.fn(),
+    },
+    initialize: jest.fn(),
+  },
+}));
+
+class MockWebSocket {
+  static inst: MockWebSocket;
+  onopen: () => void = () => { };
+  onerror: (event: any) => void = () => { };
+  onclose: (event: any) => void = () => { };
+  onmessage: (msg: any) => void = () => { };
+  close = jest.fn();
+  send = jest.fn();
+
+  constructor() {
+    MockWebSocket.inst = this;
+    setTimeout(() => {
+      this.onopen();
+    }, 10);
+  }
+}
+
+describe('WebSocketManager', () => {
```

Comment:
Here, I have mocked the `webSocket` creation in a similar manner to the way it was being done in agentx.